lib/addons/aws-loadbalancer-controller/index.ts

```diff
@@ -0,0 +1,96 @@
+import * as cdk from "@aws-cdk/core";
+import * as iam from "@aws-cdk/aws-iam";
+import { ClusterAddOn, ClusterInfo } from "../../stacks";
+import request from 'sync-request';
+
+/**
+ * Configuration options for AWS Load Balancer controller
+ */
+export interface AwsLoadBalancerControllerProps {
+    /**
+     * Namespace where controller will be installed
+     */
+    namespace?: string,
+    /**
+     * Version of the controller, i.e. v2.2.0 
+     */
+    version?: string,
+
+    /**
+     * Enable Shield (must be false for CN partition)
+     */
+    enableShield?: boolean,
+
+    /**
+     * Enable WAF (must be false for CN partition)
+     */
+    enableWaf: boolean,
+
+    /**
+     * Enable WAFV2 (must be false for CN partition)
+     */
+    enableWafv2?: boolean
+}
+
+/**
+ * Defaults options for load balancer controller.
+ */
+const AwsLoadBalancerControllerDefaults: AwsLoadBalancerControllerProps = {
+    namespace: 'kube-system',
+    version: 'v2.2.0',
+    enableShield: false,
+    enableWaf: false,
+    enableWafv2: false
+}
+
+const AWS_LOAD_BALANCER_CONTROLLER = 'aws-load-balancer-controller';
+
+export class AwsLoadBalancerControllerAddOn implements ClusterAddOn {
+
+    private options: AwsLoadBalancerControllerProps;
+
+    constructor(props?: AwsLoadBalancerControllerProps) {
+        this.options = { ...AwsLoadBalancerControllerDefaults, ...props ?? {}};
+    }
+
+    deploy(clusterInfo: ClusterInfo): void {
+
+        const cluster = clusterInfo.cluster; 
+        const awsControllerBaseResourceBaseUrl = `https://raw.githubusercontent.com/kubernetes-sigs/aws-load-balancer-controller/${this.options.version}/docs`;
+        const awsControllerPolicyUrl = `${awsControllerBaseResourceBaseUrl}/install/iam_policy${cluster.stack.region.startsWith('cn-') ? '_cn' : ''}.json`;
```

Comment:
nit...but move the url vars below service account creation. Makes it clear where the vars are used.

Reply:
Yeah, I stole this one, thought it was pretty good effort to handle cn regions as well.